### ISSUE TITLE
FooGallery Support

### ISSUE THREAD

**Issue by violence88** [mnelson4/printmyblog#12]
Hello

I've traveled a lot and have a wordpress site with many picrutres using FooGallery.
Is there any possibility to get the FooGallery running with print my blog?

In the PDF it shows the code for the Gallery - for Example:
[foogallery id="123"]

Cheers
Fabio

**Comment by mnelson4** [comment]
Thanks for the suggestion. I'll try to look into that in the next few weeks. For my reference, this is the gallery plugin: https://wordpress.org/plugins/foogallery/

**Comment by mnelson4** [comment]
FYI @violence88, my next major feature is https://github.com/mnelson4/printmyblog/issues/6, but this will be after it. There's a ton of plugins/themes with shortcodes like this that don't work, and I should probably spend some time trying to find a more general solution...

**Comment by violence88** [comment]
Hello mnelson4

Yes, this is the plugin https://wordpress.org/plugins/foogallery/
That would be awesome! I'm looking forward for the updated version! 

Thanks a lot for your Support

Cheers
Fabio

**Comment by mnelson4** [comment]
Hi Fabio, I have some progress to report!
I've created a version of Print My Blog that shows the Foo Gallery. Would you mind trying it out? https://github.com/mnelson4/printmyblog/archive/BUG/foogallery-support.zip

**Comment by violence88** [comment]
Hey mnelson4
Thank you for contacting me!

I did some testing with a local copy of my webpage. With my standard Theme (activello) Foogallery doesn't work at all with print my blog. With other Themes it does! I don't care about the theme as log i can print it :)..

Do you have an recommandation for a theme which works best with print my blog?

**Back too Foogallery:**
Single pictures are in High Quality an look awesome!
Galleries with more than one Picture are lined up and not in the actual foogallery design.
I do have many gallerys with two to three pictures.

Cheers
Fabio

**Comment by mnelson4** [comment]
Hi Fabio, thanks for giving it a whirl!
Ideally print my blog should work with all themes. I just tried it with activello and it worked for me, so there must be some setting I have different. What is on your site’s homepage? Is it the blogs page, or a different template?
Also, could you maybe send me a screenshot or a pdf copy that shows the problem with foogallery? And a screenshot of how they look when viewed normally, please (for comparison).

**Comment by violence88** [comment]
Hi mnelson4 

this is the page im talking about https://www.saraundfabio.reisen
i know i still have some work to do.. this page was once totaly fine.. after all updates some things doesnt look right (like GPS tracking...)
also i had problems with the lightbox of foobar..

(always with the setting Columns 1)

activello print:
![activello](https://user-images.githubusercontent.com/48052083/54162315-5d50dc00-4455-11e9-889f-2f8b92cc3acf.JPG)

other themes:
![other_theme](https://user-images.githubusercontent.com/48052083/54162339-75c0f680-4455-11e9-91c6-b06386b05e72.JPG)

pictures in blog:
![pictures2](https://user-images.githubusercontent.com/48052083/54162354-85d8d600-4455-11e9-85a7-e046f605087b.JPG)

pictures printed
![pictures1](https://user-images.githubusercontent.com/48052083/54162375-91c49800-4455-11e9-939b-2b2284d97679.JPG)

gallery printed
![pictures3](https://user-images.githubusercontent.com/48052083/54162394-9d17c380-4455-11e9-970d-ace011bbfcf1.JPG)

cheers fabio

**Comment by mnelson4** [comment]
hey fabio, I've updated the foogallery-integration version of the plugin. Please [re-download it](https://github.com/mnelson4/printmyblog/archive/BUG/foogallery-support.zip). 
It should fix how activello only made the post content take up half the page width (and prevent that issue for other themes that similarly made the post content only occupy half the page width).
Also, I made a totally baby-brained mistake that made the foogallery js and css not load for you (but it did work for me). I've fixed it and so hopefully the gallery will work better.
Let me know how it goes this next time! :crossed_fingers:

**Comment by violence88** [comment]
Hello mnelson4 

Thanks for the update! 

Activello works!

Unfortunalty foogally with more than one picture still looks the same.
I also created a new blogpost and made some testing with different foogallery settings.

*NEW
The "Featured Image" doesn't get printed anymore. Also tried it with different images and new blogpost.

**Comment by mnelson4** [comment]
Ah, I see what was wrong: you were using a Masonry gallery which was causing a JS error, so I need to take that into account too. The [updated version](https://github.com/mnelson4/printmyblog/archive/BUG/foogallery-support.zip) should fix that (but it's possible there will be another error after that one).
(FYI once I have a quickfix done for your needs, I'm wanting to work on a more general fix, but that will take much longer)
And thanks for catching that the featured images weren't getting displayed. I missed that.

**Comment by violence88** [comment]
I tried it with different gallery styles before. But know it seems to work.
Just the image is not showing but there so many foogallery options.. bet i will get this running.
![pictures4](https://user-images.githubusercontent.com/48052083/54238567-19bfa600-4519-11e9-8008-e8a5d2801625.JPG)


The "Featured Image" works again :)

**Comment by mnelson4** [comment]
One thing I noticed: with some settings of foogallery, the images only load once you scroll down to them on the page. It’s the same when you print an individual page normally with foo gallery.
So you may need to scroll through the entire print-page, then print it.
(By the way, I see your blog has a lot of maps. I think when there are so many of those on the same page they really slow it down. May I ask how you’re adding all those maps?)

**Comment by violence88** [comment]
I will get use of the new Filter feature and will print each Post one by one.
The maps are getting printed. I only had to zoom in a bit.

**Comment by mnelson4** [comment]
When I try to print your site, I see the webserver is refusing to load a lot of the images when they're requested in bulk like this (eg using Chrome's web inspector, this is pretty typical: https://drive.google.com/file/d/1eXfywl9sI6eCCZRbAGsocBTK4642-RJZ/view. Notice all the red, indicating images that were requested, but your server took more than 2 minutes to serve, so the browser gave up.)
You need to instruct Print My Blog to slow down. You can do this by changing the "Post Rendering Wait Time" (see https://drive.google.com/a/eventespresso.com/file/d/1mK1H4UlQx9B7oMwgDrYL9t9cO1c87RGM/view?usp=drivesdk, it's a buried setting because most users shouldn't need to worry about it). It defaults to 200 milliseconds, whereas I'm afraid your site needs more like 20000 milliseconds. That will make the print-page load at a snail's pace, but it's slow enough that the server doesn't get overloaded and the images load better. And it's probably better than manually printing each page individually.

When I tried that, the images loaded ok, but the foogallery code didn't load at the very end :weary:. I made a tweak to the [latest version](https://github.com/mnelson4/printmyblog/archive/BUG/foogallery-support.zip) to try to delay loading the foogallery code, so your server will hopefully load it. If you update again and set the "Post Rendering Wait Time" to 20000 (that's twenty thousand), hopefully it will all work.

**Comment by violence88** [comment]
Hi mnelson4

Thats why i work with my local copy :). Also the pictures on the blog are watermarked (at least most of them). The Blog was once a lot faster with the same settings. After some time and updates it feels (and is :) actually) very slow and also some features do not work anymore. It will go offline soon anyway.

I did disabled 'lazy loading' in Foogallery to made them print with print my blog. i think this is also one reasons why its loading that slow because it needs to load all the images of the blog post at one time.

At the end of each post i do have a chart of the daily footsteps which are not printed so far. i will wait and see if this works in the next official release. Otherwise i take screenshots and replace the code with a picture for the print.

Anyway - I updated the plugin but im currently on my phone and it doesnt print. Need to test it later on my pc.

cheers
fabio




Am 13.03.2019 19:24 schrieb Mike Nelson <notifications@github.com>:

When I try to print your site, I see the webserver is refusing to load a lot of the images when they're requested in bulk like this (eg using Chrome's web inspector, this is pretty typical: https://drive.google.com/file/d/1eXfywl9sI6eCCZRbAGsocBTK4642-RJZ/view<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdrive.google.com%2Ffile%2Fd%2F1eXfywl9sI6eCCZRbAGsocBTK4642-RJZ%2Fview&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905731126&sdata=oU%2FxC%2Fp9GUq6xkOFdeN6NNoSOx19l%2BRYvfxUPAS0xTk%3D&reserved=0>. Notice all the red, indicating images that were requested, but your server took more than 2 minutes to serve, so the browser gave up.)
You need to instruct Print My Blog to slow down. You can do this by changing the "Post Rendering Wait Time" (see https://drive.google.com/a/eventespresso.com/file/d/1mK1H4UlQx9B7oMwgDrYL9t9cO1c87RGM/view?usp=drivesdk<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdrive.google.com%2Fa%2Feventespresso.com%2Ffile%2Fd%2F1mK1H4UlQx9B7oMwgDrYL9t9cO1c87RGM%2Fview%3Fusp%3Ddrivesdk&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905751148&sdata=uU3qbv7jem%2BBmxdcqjETW96nl03wwzycwrzvziPecrY%3D&reserved=0>, it's a buried setting because most users shouldn't need to worry about it). It defaults to 200 milliseconds, whereas I'm afraid your site needs more like 20000 milliseconds. That will make the print-page load at a snail's pace, but it's slow enough that the server doesn't get overloaded and the images load better. And it's probably better than manually printing each page individually.

When I tried that, the images loaded ok, but the foogallery code didn't load at the very end 😩. I made a tweak to the latest version<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmnelson4%2Fprintmyblog%2Farchive%2FBUG%2Ffoogallery-support.zip&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905761159&sdata=ApTl3NzhJ2AQovohp9XXs2iyDHluwtTIdC6v0YE1lU4%3D&reserved=0> to try to delay loading the foogallery code, so your server will hopefully load it. If you update again and set the "Post Rendering Wait Time" to 20000 (that's twenty thousand), hopefully it will all work.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmnelson4%2Fprintmyblog%2Fissues%2F12%23issuecomment-472547661&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905771158&sdata=xHpozcnz%2B3aUTkxcIJLQo99KuKpwqo7eRpZD6gHGOgU%3D&reserved=0>, or mute the thread<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAt03c3vQvVrO3zuEAFsjek3kICx-i2X7ks5vWUJwgaJpZM4bUKVg&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905781163&sdata=lEq75icP16c1qywA9V9EPtbupEPziRLXoNGbxc5o1F8%3D&reserved=0>.


Am 13.03.2019 19:24 schrieb Mike Nelson <notifications@github.com>:

When I try to print your site, I see the webserver is refusing to load a lot of the images when they're requested in bulk like this (eg using Chrome's web inspector, this is pretty typical: https://drive.google.com/file/d/1eXfywl9sI6eCCZRbAGsocBTK4642-RJZ/view<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdrive.google.com%2Ffile%2Fd%2F1eXfywl9sI6eCCZRbAGsocBTK4642-RJZ%2Fview&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905731126&sdata=oU%2FxC%2Fp9GUq6xkOFdeN6NNoSOx19l%2BRYvfxUPAS0xTk%3D&reserved=0>. Notice all the red, indicating images that were requested, but your server took more than 2 minutes to serve, so the browser gave up.)
You need to instruct Print My Blog to slow down. You can do this by changing the "Post Rendering Wait Time" (see https://drive.google.com/a/eventespresso.com/file/d/1mK1H4UlQx9B7oMwgDrYL9t9cO1c87RGM/view?usp=drivesdk<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdrive.google.com%2Fa%2Feventespresso.com%2Ffile%2Fd%2F1mK1H4UlQx9B7oMwgDrYL9t9cO1c87RGM%2Fview%3Fusp%3Ddrivesdk&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905751148&sdata=uU3qbv7jem%2BBmxdcqjETW96nl03wwzycwrzvziPecrY%3D&reserved=0>, it's a buried setting because most users shouldn't need to worry about it). It defaults to 200 milliseconds, whereas I'm afraid your site needs more like 20000 milliseconds. That will make the print-page load at a snail's pace, but it's slow enough that the server doesn't get overloaded and the images load better. And it's probably better than manually printing each page individually.

When I tried that, the images loaded ok, but the foogallery code didn't load at the very end 😩. I made a tweak to the latest version<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmnelson4%2Fprintmyblog%2Farchive%2FBUG%2Ffoogallery-support.zip&data=02%7C01%7C%7C1902afc9d3ce425a839608d6a7e12e02%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636880982905761159&sdata=ApTl3NzhJ2AQovohp9XXs2iyDHluwtTIdC6v0YE1lU4%3D&reserved=0> to try to delay loading the foogallery code, so your server will hopefully load it. If you update again and set the "Post Rendering Wait Time" to 20000 (that's twenty thousand), hopefully it will all work.

**Comment by mnelson4** [comment]
> At the end of each post i do have a chart of the daily footsteps which are not printed so far.

How is that being added? I'm guessing that requires other Javascript. (I'm not sure if I should continue trying to make a hack for every plugin that uses Javascript in the post content; I was trying to just get you up-and-running quickly. Not sure if I've really succeeded!)

> I updated the plugin but im currently on my phone and it doesnt print. Need to test it later on my pc.

Yeah every browser handles printing differently, but from my limited testing phone browsers are the worst at printing, so that may be the issue. Also, your site randomly doesn't respond while loading posts, which makes Print My Blog stop loading posts. So that may have been the issue (just retrying sometimes works).

**Comment by violence88** [comment]
> How is that being added? 

yes, it's generated by a plugin called wp-charts
https://de.wordpress.org/plugins/wp-charts/

but dont focus on that - i can take a screenshot and replace the code with an image.

> Not sure if I've really succeeded!

not yet unfortunatly - i will try get it work tomorrow but first attempt failed.

> I'm not sure if I should continue trying to make a hack for every plugin that uses Javascript in the post content; I was trying to just get you up-and-running quickly. 

I cant answer this question but it seems like lot of work!

**Comment by mnelson4** [comment]
Ah kk thanks for pointing me to the charting plugin. Yeah once I have the more general fix for content requiring Javascript, I'll test it on that plugin. But I don't think I'll try to hack it.

Besides that, I'm sorry but I'm going to blame the host you're using for inconsistently serving content. It's difficult to get everything to work properly when your server isn't cooperating :slightly_frowning_face:

**Comment by mnelson4** [comment]
FYI I discovered a similar service today: bloxp.com, which is actually also totally free. I tried it on your site but it seemed to have the same issue with requests intermittently failing. Nevertheless, its results might suit you more.

Also, I'm releasing this changes as part of 1.6.0 of the plugin so other users of FooGallery can benefit from them. Thanks for the feature request and help testing.

Fixed on #15
